app/common/prescription/controllers/patientSimplifiedPrescriptionController.js

```diff
@@ -370,45 +398,97 @@
+            vm.selectedProvider =  { display: '' };
             isPrescriptionControl();
+            getProviderAndPrescriptionDateModal().modal('hide');
             spinner.forPromise(loadSavedPrescriptions(patient));
           })
           .catch(function () {
             notifier.error($filter('translate')('COMMON_MESSAGE_COULD_NOT_CREATE_PRESCRIPTION'));
           });
       }
-    }
-
-    function validateBeforeSave(prescription){
+      else{
+        vm.selectedProvider = { display: '' };
+      }
+     }
 
-      var isArvPrescriptionToBeCreated = false;
-      _.forEach(prescription.prescriptionItems, function (prescriptionItem) {
-           if(prescriptionItem.regime){
-             isArvPrescriptionToBeCreated = true;
-           }
-      });
+    function validateCreatePrescription(form, prescription){
 
-        var hasExistingArvPrescription = false;
-       _.forEach(vm.existingPrescriptions, function (prescription) {
-         _.forEach(prescription.prescriptionItems, function (prescriptionItem) {
-            if( prescription.prescriptionStatus == 'ACTIVE' && prescriptionItem.regime){
-            hasExistingArvPrescription = true;
-            }
-         });
-        });
+      if(form.selectedProvider.$invalid)
+      {
+        return false;
+      }
 
-       if(isArvPrescriptionToBeCreated && hasExistingArvPrescription){
-         notifier.error($filter('translate')('COMMON_MESSAGE_COULD_NOT_CREATE_ARV_PRESCRIPTION_BECAUSE_EXISTS_AN_ACTIVE_ARV_PRESCRIPTION'));
+      if(hasActiveArvPrescriptionForNewArvItem(prescription)){
+        notifier.error($filter('translate')('COMMON_MESSAGE_COULD_NOT_CREATE_ARV_PRESCRIPTION_BECAUSE_EXISTS_AN_ACTIVE_ARV_PRESCRIPTION'));
+        return false;
+      }
+      var duplicatedItems = [];
+      duplicatedItems = hasDuplicatedExistingActivePrescriptionItems(prescription, duplicatedItems);
+      if(!_.isEmpty(duplicatedItems)){
+        notifier.error($filter('translate')('COMMON_MESSAGE_ERROR_EXISTIS_ACTIVE_PRESCRIPTIONS_FOR_ENTERED_ITEMS', {EXISTING_ITEM: _.flatMap(duplicatedItems, 'drugOrder.drug.display').toString()}));
+        return false;
+      }
+      return true;
+    }
 
-         return false;
-       }
+    function hasActiveArvPrescriptionForNewArvItem(prescription){
+      var exists = false;
+      _.forEach(prescription.prescriptionItems, function (newPrescriptionItem) {
+        if(newPrescriptionItem.regime){
+          _.forEach(vm.existingPrescriptions, function (existingPrescription) {
+            _.forEach(existingPrescription.prescriptionItems, function (prescriptionItem) {
+              if(existingPrescription.prescriptionStatus === 'ACTIVE' && prescriptionItem.regime){
+                exists = true;
+                return;
+              }
+            });
+          });
+        }
+      });
+      return exists;
+    }
+    function hasDuplicatedExistingActivePrescriptionItems(prescription, returnResult) {
```

Comment:
Not sure what this function is supposed to be returning. If it is Boolean it should return true/false, if Array empty/list of duplicated items.

Reply:
I think the argument returnResult is not necessary.